src/sql/src/ast/transform.rs

```diff
@@ -13,18 +13,121 @@ use std::collections::{BTreeMap, BTreeSet};
 
 use mz_ore::str::StrExt;
 use mz_repr::GlobalId;
-use mz_sql_parser::ast::CreateWebhookSourceStatement;
 
 use crate::ast::visit::{self, Visit};
 use crate::ast::visit_mut::{self, VisitMut};
 use crate::ast::{
     AstInfo, CreateConnectionStatement, CreateIndexStatement, CreateMaterializedViewStatement,
     CreateSecretStatement, CreateSinkStatement, CreateSourceStatement, CreateTableStatement,
-    CreateViewStatement, Expr, Ident, Query, Raw, RawItemName, Statement, UnresolvedItemName,
-    ViewDefinition,
+    CreateTypeStatement, CreateViewStatement, CreateWebhookSourceStatement, Expr, Ident, Query,
+    Raw, RawItemName, Statement, UnresolvedItemName, ViewDefinition,
 };
 use crate::names::FullItemName;
 
+/// Given a [`Statement`] rewrites all references of the schema name `cur_schema_name` to
+/// `new_schema_name`.
+pub fn create_stmt_rename_schema_refs(
+    create_stmt: &mut Statement<Raw>,
+    cur_schema_name: &str,
+    new_schema_name: String,
+) {
+    let maybe_update_item_name = |item_name: &mut UnresolvedItemName| {
+        if let [.., schema, _item] = &mut item_name.0[..] {
+            if schema.as_str() == cur_schema_name {
+                *schema = Ident::new(new_schema_name.clone());
+            }
+        }
```

Comment:
Should the else branch here be an error? When would a name ever not be fully qualified?

Reply:
Tbh I'm not sure. Currently since `UnresolvedItemName` is just a collection of idents, so there is no type-level guarantee this will be fully qualified. There is an issue (https://github.com/MaterializeInc/materialize/issues/5591) to express how well qualified an ident is.

If something is not fully qualified then it's probably okay? No need to rename anything then?

Reply:
> If something is not fully qualified then it's probably okay? No need to rename anything then?

Yeah that's a good point. I don't think we have to worry about it in this PR.